test/thread/TestKanoWorldSave.h

```diff
@@ -0,0 +1,68 @@
+/**
+ * \file TestKanoWorldSave.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief Tests for the Mercury::KanoWorld save and load concurrency
+ *
+ */
+
+#ifndef TEST_THREAD_TESTKANOWORLDSAVE_H_
+#define TEST_THREAD_TESTKANOWORLDSAVE_H_
+
+#include <gmock/gmock.h>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <chrono>  // NOLINT
+#include <functional>
+#include <random>
+#include <string>
+#include <thread>  // NOLINT
+#include <utility>
+#include <vector>
+
+#include "mercury/kw/kw.h"
+#include "test/fixtures/randomisation.h"
+
+
+using testing::Eq;
+
+
+class KWSaveThread : public Mercury::KanoWorld::KanoWorld,
+                     public testing::Test {
+    // Empty - bridge to protected members for unit-testing
+};
+
+
+/**
+ * \warning Currently this only checks that we don't end up in deadlocks, in
+ *          the future we would want to test that the saved data is consistent
```

Comment:
I guess the catch is that the only way we'd spot a deadlock is if the tests end up in an infinite loop? If so, it'd be nice to at least add a note about the need for a deadlock detector.